test/linalg/symmetric.jl

```diff
@@ -42,206 +41,229 @@ end
-            @test Hermitian(asym) * Hermitian(asym) ≈ asym*asym
-            @test_throws DimensionMismatch Hermitian(asym) * ones(eltya,n+1)
-            Base.LinAlg.A_mul_B!(C,a,Hermitian(asym))
-            @test C ≈ a*asym
-        end
-        if eltya <: Real
-            @test Symmetric(asym) * Symmetric(asym) ≈ asym*asym
-            @test Symmetric(asym) * a ≈ asym * a
-            @test a * Symmetric(asym) ≈ a * asym
-            @test_throws DimensionMismatch Symmetric(asym) * ones(eltya,n+1)
-            Base.LinAlg.A_mul_B!(C,a,Symmetric(asym))
-            @test C ≈ a*asym
-        end
-        tri_b = UpperTriangular(triu(rand(eltya, n, n)))
-        @test Array(Hermitian(asym).' * tri_b) ≈ asym.' * Array(tri_b)
-        @test Array(tri_b * Hermitian(asym).') ≈ Array(tri_b) * asym.'
-        @test Array(Hermitian(asym)' * tri_b) ≈ asym' * Array(tri_b)
-        @test Array(tri_b * Hermitian(asym)') ≈ Array(tri_b) * asym'
-
-        # solver
-        @test Hermitian(asym)\x ≈ asym\x
-        if eltya <: Real
-            @test Symmetric(asym)\x ≈ asym\x
+            @testset "transpose, ctranspose" begin
+                S = Symmetric(asym)
+                H = Hermitian(asym)
+                if eltya <: Real
+                    @test  transpose(S) === S == asym
+                    @test ctranspose(S) === S == asym
+                    @test  transpose(H) === H == asym
+                    @test ctranspose(H) === H == asym
+                else
+                    @test  transpose(S) === S
+                    @test ctranspose(S) ==  Symmetric(conj(asym))
+                    @test  transpose(H) ==  Hermitian(transpose(asym))
+                    @test ctranspose(H) === H == asym
+                end
+            end
         end
 
-        #inversion
-        @test inv(Hermitian(asym)) ≈ inv(asym)
-        if eltya <: Real && eltya != Int
-            @test inv(Symmetric(asym)) ≈ inv(asym)
-            @test inv(Hermitian(a)) ≈ inv(full(Hermitian(a)))
-            @test inv(Symmetric(a)) ≈ inv(full(Symmetric(a)))
-        end
+        @testset "linalg unary ops" begin
+            # Revisit when implemented in julia
+            if eltya != BigFloat
+                @testset "trace" begin
+                    @test trace(asym) == trace(Hermitian(asym))
+                end
+                @testset "cond" begin
+                    @test cond(Hermitian(asym)) ≈ cond(asym)
+                end
+                
+                @testset "inversion" begin
+                    @test inv(Hermitian(asym)) ≈ inv(asym)
+                    if eltya <: Real && eltya != Int
+                        @test inv(Symmetric(asym)) ≈ inv(asym)
+                        @test inv(Hermitian(a)) ≈ inv(full(Hermitian(a)))
+                        @test inv(Symmetric(a)) ≈ inv(full(Symmetric(a)))
+                    end
+                end
+
+                @testset "symmetric eigendecomposition" begin
```

Comment:
Perhaps this should actually be moved to `test/linalg/eigen.jl` since these tests are not specific to `Symmetric`  /`Hermitian`?